### ISSUE TITLE
update/remove CHEF-KOCH lists now offline

### ISSUE THREAD

**Issue by collinbarrett** [collinbarrett/FilterLists#584]
per https://github.com/collinbarrett/FilterLists/issues/574#issuecomment-430443843 and https://github.com/collinbarrett/FilterLists/issues/574#issuecomment-430443843

**Comment by DandelionSprout** [comment]
Apparently this is the thread that explains the situation: https://www.reddit.com/r/uBlockOrigin/comments/9opkel/chefkoch/

I'm not particularly comfortable with this turn of events. If links to illegal material are now permabannable material, how are list maintainers supposed to add them to adblock lists then? 😬

**Comment by collinbarrett** [comment]
Oof, Well, that sucks. Yeah, his lists are designed to BLOCK those sites. GitHub doesn't take into context what the project is? Man...

When the MS announcement about buying GH was first made and the initial wave of maintainers jumped ship, I ignored it, as MS is turning around as a company. Huge fan of their dedication to open source.

Not blaming MS for the current policies, but, boy, that seems super heavy-handed. Making me wonder about staying on GH...

Let's put this issue on hold for a few days to see what shakes out. Maybe they will reconsider and GH will bring CHEF-KOCH back online...

**Comment by DandelionSprout** [comment]
CHEF-KOCH mentioned on Reddit that he has a (part-time?) lawyer, which is more than what I can boast of having. We can only hope that his lawyer can squeeze out a proper answer from GitHub about it, before GitHub falls out of favour with the adblocking community.

It could be the Onion lists, it could be that he had an entry in an anti-porn list that went beyond merely being about porn, or it could be something that wasn't about the adblock lists at all. Therefore I cross my fingers that a clarification is given from GitHub about this before long.

**Comment by DandelionSprout** [comment]
Okay, so there's some more details that have begun to emerge. Apparently there was a hatedom among a group of Firefox privacy enthusiasts, that wanted to see him burn: https://goo.gl/rA99aX

Forgive me for using a link shortener, but I'm getting some serious 'Steven Universe fandom'-style vibes from those guys, so I really don't want them to know about this very thread out of fear of what they could try to do to us out of whatever reason.

**Comment by DandelionSprout** [comment]
I **could** enter the lion's mouth sometime tomorrow, by asking them if they have know of any other lists that were sourced from other lists. But that'd require me to do more mental preparation than I can do tonight.

**Comment by DandelionSprout** [comment]
Additionally, **if** it turns out that one of those enthusiasts filed a false report against CHEF-KOCH to get him taken down, which may or may not be the case, then I don't want to go down in history as being someone who were on their side.

They are already apparently targeting his GitLab account, for currently unknown purposes.

**Comment by DandelionSprout** [comment]
I've been thinking a bit more. I have been known to be a bit paranoid, pessimistic, and for not wanting to tempt fate, which may influence these viewpoints of mine quite a bit. And in the hopefully avoidable scenario where either party involved learns about this thread (or about FilterLists.com for that matter), I'd like to point out to them that I am **not** eligible to speak on behalf of FilterLists.com, and most probably never have.

The thing is, I'm not sure which of those two I should believe. CK has glaring gaps in his explanations on Reddit (How does people get "censored" on non-GitHub git sites? And what happened the first seven times he got suspended?), but has kept his face more or less straight. Whereas the enthusiasts are acting like drunk football fans after a cup final, and have manhunted him through multiple social venues for a while, but they have collected a fairly large number of incidents to look closer into (but unfortunately only 2 of the incidents concern any of his filterlists).

This is a case where it'd pay off big time if we had any real grasp of the situation; but the two of them seem to have been at odds with each other for years, and there's even a chance that they'll be going to court against each others. So I think the feelings could run really high if we were to inquire either site about it. Conversely, by *not* asking them, and instead just taking a decision on our own, I see no less than three worst-case scenarios that could unfold:

**Worst case scenario 1:** By showing unbridled support for the enthusiasts, there's a miniscule (but very unlikely) chance that CK could take insult with us as well for removing his lists without legally verified proof. And he kinda has a lawyer. And from the looks of it he has the money to run legal proceedings as well.

**Worst case scenario 2:** In the (increasingly unlikely) event that the reporter (The identity of which is not known ATM) gets off from this event scot-free, and presuming that (s)he did it out of spite, (s)he could very well feel empowered and willing to try to take down other accounts (s)he doesn't like by falsely accusing them of hosting illegal links. It could even encourage other spiteful or trolling people to do the same, even those who have no connections to this case.

**Even worse case scenario:** Some random worker at an аnti-аdblоck firm sees this thread and thinks he can get a pay hike by reporting every adblock repo imaginable for hosting illegal content, even those that don't even *remotely* make use of such links in their lists. If GitHub doesn't raise any eyebrows at such topical mass-reports, and since list mirrors have not been added to uBO in droves yet, it'd potentially lead to >400 lists and thousands of side-projects gone overnight, and the beginning of a cyberwar that humanity has never seen the likes of before.

**Comment by Atavic** [comment]
He recently criticized NetGuard - no-root firewall for android. I found his points of great value.

**Comment by KonoromiHimaries** [comment]
> I'm not particularly comfortable with this turn of events. If links to illegal material are now permabannable material, how are list maintainers supposed to add them to adblock lists then? 😬

https://github.com/NanoAdblocker/NanoCore/issues/180

**Comment by KonoromiHimaries** [comment]
> that seems super heavy-handed. Making me wonder about staying on GH...

your #595 issue it disappeared due to the last crash of github
m$, of course, did not send an e-mail with info about the crash

**Comment by DandelionSprout** [comment]
@KonoromiHimaries GitHub slammed hard into the gutter yesterday, but thankfully they were able to recover code pushes and pulls that were stuck in the void.

I've seen experienced coders claim that GitHub has had similar crashes in the past, but GH could've at least had the dignity to add an information banner for logged-in users or something like that while the problems were going on.

**Comment by Atavic** [comment]
A link to *status.github.com* is provided on bottom page.

**Comment by Thorin-Oakenpants** [comment]
Hi guys. I'm the owner of the ghacks user.js repo. Apologies if this get a wee bit long.

Regardling CK. I have a "little" history with him. I say "little" because I try to avoid anything to do with him. 3 to 4 years ago  I stopped contributing ideas at PK's user.js rep (https://github.com/pyllyukko/user.js) due to his tone/manner (I just attributed it to that being his style). I had my own independent project anyway (updated in articles at ghacks.net from time to time - hence the name). I eventually moved it to github in early 2017. Before than and for a while afterwards, I was a pretty frequent commentator in ghacks articles on Firefox. At some stage, out of the blue, CK just went all anti-me and our project (https://www.ghacks.net/2017/08/22/firefox-55-new-preferences-and-ghacks-user-js-changes/#comments). Up until this point I had forgotten about him. After a few comments it came to light that he was being hypocritical (on the user.js contents) and basically copying our work/research - as well as, IMO, being totally ingenuous and ignorant of so many facts. So I decided to ban him from my repo as a preventative measure. And once again, I ignored him. It's not worth my time to try and reason with him.

Then it came to light that he started an anti-ghacks user.js (which I started as a topic to set the record straight on the factual inaccuracies) but it quickly emerged that he had copied some of our work (one of our guys created a batch file which he totally copied with attribute). This led, over time, to the discovery of other work also being copied. I'm talking about the first post where 3 projects (not filter lists) were copied, and the author changed to CK, with no acknowledgement of the original creator. In our case it also involved changing the licence. I also pointed out one filter list in later comments.

In all the time since the first verbal attacks on us/me in ghack.net comments, to the plagiarizing of our work (the batch file), to the discovery by others of plagiarized work (ASN_IPFire_Script , Lanchon/REPIT) I and our community have ignored him. I guess I can't really speak for others, and the thread is old and long, and as you correctly say, a bit shouty with blodthirst at times.

I will state once again: [source](https://github.com/ghacksuserjs/ghacks-user.js/issues/323#issuecomment-431627446)
> I will state this, for the record: I have nothing to do with CK being blocked/banned on GitHub, both this incident and all other previous occasions. I actually go out of my way to avoid CK (banning him from this repo, stopped making comments at ghacks, etc) because in my life, I try to cut out the negativity, the time-wasting, and the bullies/abusers/loud-mouthed and the unreliable (you only ever get to let me down once). That's in real life, but I extend this to the web in general. And, I have never posted any comments on reddit (and probably never will). I am not crssi or SkcusYllaerTidder or anyone else.

Quiet simply put, all of CK's problems are his own doing IMO. It certainly isn't anything to do with me. I totally respect the work people do in creating and maintaining projects, including the invaluable work of filter lists. But I don't respect plagiarizing and passing off others' work as your own. CK's actions made him suspect. I wouldn't have relied on him if my life depended on it. I too can only speculate on why he was taken down by GitHub - the guy lies too much to know what is really going on. For all I know, it was yet another party he ripped work off. 8th time being banned? IDK and I don't want to know. As for all his "lawyer" talk, it's all smoke and mirrors - he threatens everyone with that. He even tried to get uBlock Origin banned! Draw your own conclusions.

Long story short, you have nothing to fear from me. I'm a rational, sane human being who likes to provide better privacy and security solutions free to the world, and work with teams of people. I've never had a "beef" with anyone (see my quote above, I just stop letting them in my life and move on). Don't let the paranoia, or false words of CK, seep in. Chill out :) And just put measures into place to mitigate this kind of train-wreck in future.

Keep up the good work :kiss:

**Comment by DandelionSprout** [comment]
Thank you for the very detailed and reasonable explanation. 👍

While I personally have been secretly waiting out to see if more claims of plagiarism emerged before I'd pick a stance on all this, it's nice to see that you seem like a good enough guy to me, at least.

>As for all his "lawyer" talk, it's all smoke and mirrors - he threatens everyone with that. He even tried to get uBlock Origin banned! Draw your own conclusions.

Shit. Okay, now that sounds pretty problematic of him to try to accomplish, from the sounds of it.

**Comment by claustromaniac** [comment]
> Whereas the enthusiasts are acting like drunk football fans after a cup final

That made me chuckle.

Your doubts make total sense to me. I think pretty much anyone who regards that guy with respect and then finds that thread will likely get similar vibes from it. What happens is we (the ones celebrating) have long lost the respect for him. In fact, I never had any respect for him in the first place, because I didn't even know he existed until Pants discovered that he had plagiarized my work (and @earthlng's, who helped me a bunch). In my eyes, him getting banned is like the fall of a petty villain, but I understand that others don't share my view and I respect that. I personally don't need to convince anyone; I'm quite happy knowing the guy is figuratively behind bars for the time being. Besides, he may have been genuinely helpful to others (I honestly don't know that), so I am in no position to judge anyone else.

That being said, there is a simpler explanation: we're 24/7 drunk football fans waiting for any reason to celebrate.

By the way, very nice project. :thumbsup: I hope you don't mind having a new lurker around.


<sub>Now a somewhat off-topic shameless reply to @Thorin-Oakenpants (AKA the real reason I'm leaving a comment here, but shh don't tell anyone)</sub>

<sub>DAFUQ Pants:heavy_exclamation_mark::heavy_exclamation_mark::heavy_exclamation_mark: I know GitHub says the thread is locked to collaborators but I don't seem to qualify. I was planning to leave the topic alone anyway (had enough fun poking the guy's corpse for now), but you didn't let me reply to you! **NOT FAIR! CENSORSHIP! I'LL HAVE TO TAKE DOWN YOUR PROJECT NOW!** </sub>

> <sub>claustromaniac needs to work on getting his followers count up from 5</sub>

<sub>I'm pretty comfy staying a random guy with a single digit of total followers, TYVM.</sub>


/lurking mode ON
